Ruby Languages and frameworks (#388)
gitpod-io/website: src/docs/ruby-in-gitpod.md

# Ruby

It's relatively easy to set up your Ruby project in Gitpod.

## Examples
However, before we begin, here are some examples of Ruby projects automated with Gitpod.

Name | Open In Gitpod | Description
---------|----------|---------
 [dev.to](https://github.com/thepracticaldev/dev.to) | [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/thepracticaldev/dev.to) | a platform where software developers write articles
 [home-assistant.io](https://github.com/home-assistant/home-assistant.io) | [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/home-assistant/home-assistant.io) | Open source home automation

## VSCode Extensions

### [Ruby Test Explorer](https://marketplace.visualstudio.com/items?itemName=connorshea.vscode-ruby-test-adapter)
With the Ruby test explorer you can run unit tests from within the Gitpod UI
![](./images/ruby_test_ui.png)

> To add this extension to your repository, simply add these lines to your [`.gitpod.yml`](https://www.gitpod.io/docs/41_config_gitpod_file/) configuration file:

Reviewer: This has changed now (old links still work, but it's nicer to use new ones when possible):

```suggestion
To add this extension to your repository, simply add these lines to your [`.gitpod.yml`](https://www.gitpod.io/docs/config-gitpod-file/) configuration file:
```

```YAML
vscode:
  extensions:
    - connorshea.vscode-ruby-test-adapter@0.6.1:HO9rpcRv7bmRIuo7Mty/zg==
    - hbenl.vscode-test-explorer@2.15.0:koqDUMWDPJzELp/hdS/lWw==
```

## Ruby Version

> By default Gitpod installs Ruby v2.6.3 to change the default version check in a `.ruby-version` file into the repository with the desired version. You can also use `rvm` to manage your Ruby Version.

Reviewer: I think this should be all the way at the top, before examples or VS Code extensions.

Also, it seems to me that both Ruby 2.5 and 2.6 are pre-installed: 

https://github.com/gitpod-io/workspace-images/blob/c90b4a5ce44b8956cfd278f67422c739e07ff80e/full/Dockerfile#L221-L228

And finally, actual example commands showing how to use `rvm` would be super helpful.

Here is a suggestion:

```suggestion
Gitpod comes with Ruby 2.5 and 2.6 pre-installed by default (2.6 is the default).

To change the default Ruby version, you can simply run `rvm use 2.5 --default` in Gitpod's Terminal. You can also install other versions, e.g. by running `rvm install 2.7`.
```

## Popular Frameworks in Gitpod

### Ruby On Rails

> So, you want to write your cool new Ruby On Rails application in Gitpod? Well, here is how to do it but first to give you an idea of what to expect, please take a look at our minimal example Rails application running in Gitpod [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/JesterOrNot/Gitpod-Ruby-On-Rails)

Reviewer: More breathing opportunities:

```suggestion
So, you want to write your cool new Ruby On Rails application in Gitpod? Well, here is how to do it.

 But first, to give you an idea of what to expect, please take a look at our minimal Rails example running in Gitpod [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/JesterOrNot/Gitpod-Ruby-On-Rails)
```

# Ruby

It's relatively easy to set up your Ruby project in Gitpod.

## Examples

> However, before we begin, here are some examples of Ruby projects automated with Gitpod.

Reviewer: The "However" feels unnecessary, and slows down the reader:

```suggestion
Before we begin, here are some examples Ruby projects automated with Gitpod:
```

> Name | Open In Gitpod | Description

Reviewer: ```suggestion
Name | Description | Try it
```

---------|----------|---------
 [dev.to](https://github.com/thepracticaldev/dev.to) | [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/thepracticaldev/dev.to) | a platform where software developers write articles
 [home-assistant.io](https://github.com/home-assistant/home-assistant.io) | [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/home-assistant/home-assistant.io) | Open source home automation

> ## VSCode Extensions

Reviewer: It feels like we're missing some explanation paragraph below this title. Suggestion:

```suggestion
## VSCode Extensions

Here are a few useful extensions that you'll likely want to install in your Ruby project.
```

### [Ruby Test Explorer](https://marketplace.visualstudio.com/items?itemName=connorshea.vscode-ruby-test-adapter)

> With the Ruby test explorer you can run unit tests from within the Gitpod UI

Reviewer: Suggestion to allow readers to breathe 😅 

```suggestion
With the Ruby test explorer, you can run unit tests from within the Gitpod UI.
```

# Ruby in Gitpod

It's relatively easy to set up your Ruby project in Gitpod.

> ## Ruby Version

Reviewer: ```suggestion
## Ruby Versions
```

> Gitpod comes with Ruby 2.5 and 2.6 pre-installed by default (2.6 is the default).

Reviewer: ```suggestion
Gitpod comes with Ruby 2.5 and 2.6 pre-installed (2.6 is the default).
```

To change the default Ruby version, you can simply run `rvm use 2.5 --default` in Gitpod's Terminal. You can also install other versions, e.g. by running `rvm install 2.7`.

## Examples

> Before we begin, here are some examples Ruby projects automated with Gitpod:

Reviewer: ```suggestion
Before we begin, here are some example Ruby projects that are automated with Gitpod:
```

> Name | Description | Try it

Reviewer: ```suggestion
Repository | Description | Try it
```

---------|----------|---------

>  [dev.to](https://github.com/thepracticaldev/dev.to) | a platform where software developers write articles | [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/thepracticaldev/dev.to)

Reviewer: ```suggestion
 [dev.to](https://github.com/thepracticaldev/dev.to) | A platform where software developers write articles | [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/thepracticaldev/dev.to)
```

> # Ruby

Reviewer: I've started changing these titles to `{X} in Gitpod`, to clarify what this page is for.

```suggestion
# Ruby in Gitpod
```

It's relatively easy to set up your Ruby project in Gitpod.

## Examples
Before we begin, here are some examples Ruby projects automated with Gitpod:

Name | Description | Try it
---------|----------|---------

>  [dev.to](https://github.com/thepracticaldev/dev.to) | [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/thepracticaldev/dev.to) | a platform where software developers write articles

Reviewer: ```suggestion
 [dev.to](https://github.com/thepracticaldev/dev.to) | a platform where software developers write articles | [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/thepracticaldev/dev.to)
```

>  [home-assistant.io](https://github.com/home-assistant/home-assistant.io) | [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/home-assistant/home-assistant.io) | Open source home automation

Reviewer: ```suggestion
 [home-assistant.io](https://github.com/home-assistant/home-assistant.io) | Open source home automation | [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/home-assistant/home-assistant.io)
```

## VSCode Extensions

Here are a few useful extensions that you'll likely want to install in your Ruby project.

### [Ruby Test Explorer](https://marketplace.visualstudio.com/items?itemName=connorshea.vscode-ruby-test-adapter)
With the Ruby test explorer, you can run unit tests from within the Gitpod UI.
![](./images/ruby_test_ui.png)
To add this extension to your repository, simply add these lines to your [`.gitpod.yml`](https://www.gitpod.io/docs/config-gitpod-file/) configuration file:
```YAML
vscode:
  extensions:
    - connorshea.vscode-ruby-test-adapter@0.6.1:HO9rpcRv7bmRIuo7Mty/zg==
    - hbenl.vscode-test-explorer@2.15.0:koqDUMWDPJzELp/hdS/lWw==
```

## Ruby Version
Gitpod comes with Ruby 2.5 and 2.6 pre-installed by default (2.6 is the default).

> To change the default Ruby version, you can simply run `rvm use 2.5 --default` in Gitpod's Terminal. You can also install other versions, e.g. by running `rvm install 2.7`.

Reviewer: Please move this entire section up, at least above VS Code extensions.

Author: Here

## Ruby On Rails

> So, you want to write your cool new Ruby On Rails application in Gitpod? Well, here is how to do it.

Reviewer: You're saying "here is how to do it", but then you're not saying how to do it.

Please either add the missing explanations, or remove "Well, here is how to do it." and "But first,".

>  But first, to give you an idea of what to expect, please take a look at our minimal Rails example running in Gitpod.

Reviewer: ```suggestion
But first, to give you an idea of what to expect, please take a look at our minimal Rails example running in Gitpod:
```

# Ruby

It's relatively easy to set up your Ruby project in Gitpod.

## Examples
However, before we begin, here are some examples of Ruby projects automated with Gitpod.

Name | Open In Gitpod | Description
---------|----------|---------
 [dev.to](https://github.com/thepracticaldev/dev.to) | [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/thepracticaldev/dev.to) | a platform where software developers write articles

>  [home-assistant](https://github.com/https://github.com/gitpod-io/home-assistant) | [![Open in Gitpod](https://gitpod.io/button/open-in-gitpod.svg)](https://gitpod.io/#https://github.com/gitpod-io/home-assistant) | Open source home automation

Reviewer: The GitHub URL is broken.

Also, the project https://github.com/gitpod-io/home-assistant is 100% Python. Why add it to the Ruby guide? (I believe you want https://github.com/home-assistant/home-assistant.io instead)

Author: Which link is broken?

Reviewer: This was fixed since. Many thanks!
